### PR TITLE
RDoc-934: Client API > Commands > Patches > How to work with patch re…

### PR DESCRIPTION
**Pull request by maximburyak**
…quests

 RDoc-935: Client API > Commands > Patches > How to use JavaScript to patch your documents

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/commands/patches/how-to-work-with-patch-requests.markdown [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Commands : Patches: How to work with patch requests?
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/commands/patches/how-to-work-with-patch-requests.markdown]: This article should be removed/moved, right?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Patches: How to work with patch requests?
+
+**Patch** operation is used to perform partial document updates without having to load, modify, and save a full document. This is usefull for performance enhancement or for updating denormalized data in entities.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: typo: usefull => useful
usefull for performance enhancement => useful as a performance enhancement (not 100% sure)

Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: Maybe:
This is usefull for performance enhancement or for updating denormalized data in entities =>
The whole operation is executed on server-side and is useful as a performance enhancement or for updating denormalized data in entities

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    The current page deals with patch operations on single docuements, the next pages in this section will deal with patches on more than one docuemnt.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: typo: docuements => documents
typo: docuemnt => document

what is a page in this context? Not sure what was the intention here...

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+### 2. Non-typed session API
+A non-typed session interface that exposes the full functionality and uses the session facilities to perform multiple operations in one request.
+
+{CODE patch_non_generic_interface_in_session@ClientApi\Operations\Client\Patches\PatchRequests.cs /}
+
+**Parameters**
+
+{CODE patch_command_data@Common.cs /}
+
+We highly recommend using scripts with parameters, that allows RavenDB to cache scripts and therefor enhance performance
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: enhance => boost ?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+### 3. Operations API
+* An operations interface that exposes the full functionality and allows performing ad-hoc patch operations, without creating a session
+
+{CODE patch_non_generic_interface_in_store@ClientApi\Operations\Client\Patches\PatchRequests.cs /}
+
+**Parameters**
+
+{CODE patch_operation@Common.cs /}
+
+## Builtin javascript enhancements
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: typo: Builtin => Built-in
typo: JavaScript

Maybe enhancements => extensions?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+In addition to Ecmascript 5.1 api, RavenDB introduces few builtin functions and members:
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: typo: Ecmascript => ECMAScript
typo: api => API
typo: builtin => built-in

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+| ------ |:------:| ------ |
+| `id(document)` | function | returns the ID of a document|
+| `this` | object | Current document (with metadata) |
+| `load(key)` | method | Allows document loading, increases maximum number of allowed steps in script. See `Raven/AdditionalStepsForScriptBasedOnDocumentSize` [here](../../../server/configuration/configuration-options#javascript-parser). |
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: Raven/AdditionalStepsForScriptBasedOnDocumentSize is this configuration option valid?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+| `put(key, data, metadata)` | method | Allows document putting, returns generated key |
+| `output(...)` | method | Allows debug your patch, prints passed messages in output tab |
+
+## Eexamples
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: typo: Eexamples => Examples

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+{PANEL: Change field's value }
+{CODE-TABS}
+{CODE-TAB:csharp:Session-typed-syntax patch_firstName_generic@ClientApi\Operations\Client\Patches\PatchRequests.cs /}
+{CODE-TAB:csharp:Session-syntax-untyped patch_firstName_non_generic_session@ClientApi\Operations\Client\Patches\PatchRequests.cs /}
+{CODE-TAB:csharp:Operations-syntax patch_firstName_non_generic_store@ClientApi\Operations\Client\Patches\PatchRequests.cs /}
+{CODE-TABS/}
+{PANEL/}
+
+{PANEL: Change two fields values }
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: I think it should be (depending on intent)
1. Change two field values
2. Changes values of two fields

File: Documentation/4.0/Raven.Documentation.Pages/client-api/commands/patches/how-to-work-with-patch-requests.markdown (hunk 1 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/commands/patches/how-to-work-with-patch-requests.markdown]: ow. yeah, you are right!

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 2 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 3 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: I meant literally this documentation page (article). I also wasn't sure what's better to write here,  can you propose something else? I is important to make it clear for the readers that there are two kinds of patch operations - single document and set based

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 4 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 5 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: yes

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 6 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 7 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: No, it's not anymore, removet that remark

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 8 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 9 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: Yeah, I'll choose the second one

**Comment by maximburyak** [comment]
closing this PR, please see next one here: https://github.com/ravendb/docs/pull/365

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown (hunk 3 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/patches/how-to-work-with-patch-requests.markdown]: Maybe 'current section' or 'next section'? I'm assuming both of these patching kinds will end-up in this article (same page), right? This is why 'page' is not a best word here.

**Pull request closed without merging**